dev/release/verify-release-candidate.bat

```diff
@@ -112,6 +112,8 @@ cmake -G "%GENERATOR%" ^
       -DARROW_WITH_ZSTD=ON ^
       -DCMAKE_BUILD_TYPE=%CONFIGURATION% ^
       -DCMAKE_INSTALL_PREFIX=%ARROW_HOME% ^
+      -DCMAKE_PREFIX_PATH=%CONDA_PREFIX% ^
```

Comment:
It seems that conda's OpenSSL is installed into `%CONDA_PREFIX%\Library\lib`, `%CONDA_PREFIX%\Library\include` and so on:

```suggestion
      -DCMAKE_PREFIX_PATH=%CONDA_PREFIX%\Library ^
```